src/sentry/configurations/docs/api.md

```diff
@@ -0,0 +1,78 @@
+# Configurations API
+
+Host: https://sentry.io/api/0
+
+**Authors.**
+
+@cmanallen
+
+## Configuration [/projects/<organization_id_or_slug>/<project_id_or_slug>/keys/<key_id>/configuration/]
+
+### Get Configuration [GET]
+
+Retrieve the DSN's configuration.
+
+**Attributes**
+
+| Column             | Type   | Description                     |
+| ------------------ | ------ | ------------------------------- |
+| id                 | string | Client key.                     |
+| sample_rate        | number | A number value between 0 and 1. |
+| traces_sample_rate | number | A number value between 0 and 1. |
+| user_config        | any    | Arbitrary user supplied JSON.   |
+
+**If an existing configuration exists**
+
+- Response 200
+
+  ```json
+  {
+    "data": {
+      "id": "99aabf0dad1c48ad8e47e2a43969f312",
+      "sample_rate": 1.0,
+      "traces_sample_rate": 0.5,
+      "user_config": {
```

Comment:
Probably another dumb question,

I'm using the term “rule” to describe the action when a user defines configuration in our UI.

Using `"sample_rate": 1.0` as an example, when a user wants to define a custom config rule, will each rule expect a single key and a single value? Or could one rule have multiple keys and values pointing to variables inside the users' application?

I'm trying to understand the mental model here

Reply:
All of the keys are hard-coded.  There will be a form with these three hard-coded inputs.  If we want more inputs we would hard-code more form elements and update the protocol.

User-defined key, value pairs are not scoped yet and will come after the initial remote-configuration work.

Reply:
So, am I hearing that if a user wants to define any custom config for their application, it will be defined by a single rule?

Reply:
For the initial internal release that is true.  For the private alpha we will allow the user to define multiple rules.